File: 0007-reverse-integer/0007-reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        long long ans = 0;  
        while (x != 0) {
            int digit = x % 10;
            ans = ans * 10 + digit;
            x /= 10;

            
            if (ans > 2147483647 || ans < -2147483648) {
                return 0;
            }
        }
        return (int)ans;
    }
};

File: 0007-reverse-integer/0007-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        res = 0

        if x > 0:
            while x > 0:
                res = (res * 10) + (x % 10)
                x //= 10
        else:
            x *= -1
            while x > 0:
                res = (res * 10) + (x % 10)
                x //= 10
            res *= -1

        if res > 2**31 - 1 or res < -2**31:
            return 0

        return res